plugins/woocommerce/includes/class-wc-session-handler.php

```diff
@@ -496,24 +496,48 @@ private function is_session_expiring() {
 	 * For guests, sessions expire in 48 hours.
 	 */
 	public function set_session_expiration() {
-		$expiring_seconds   = DAY_IN_SECONDS;
-		$expiration_seconds = is_user_logged_in() ? WEEK_IN_SECONDS : 2 * DAY_IN_SECONDS;
-
 		/**
 		 * Filters the session expiration.
 		 *
 		 * @since 5.0.0
 		 * @param int $expiration_seconds The expiration time in seconds.
 		 */
-		$this->_session_expiring = time() + intval( apply_filters( 'wc_session_expiring', $expiring_seconds ) );
-
+		$expiring_seconds = intval( apply_filters( 'wc_session_expiring', DAY_IN_SECONDS ) );
+
+		if ( $expiring_seconds > MONTH_IN_SECONDS - DAY_IN_SECONDS ) {
```

Comment:
How about storing the maximum session duration in a variable to make the readability and understanding better?

```
		$max_session_duration = MONTH_IN_SECONDS; 
		$max_expiring_duration = $max_session_duration - DAY_IN_SECONDS; 
```